testrunner-running/src/main/scala/edu/illinois/cs/statecapture/StateCapture.java

```diff
@@ -0,0 +1,755 @@
+                        OFME.printStackTrace();
+                        continue;
+                    } catch (NoClassDefFoundError NCDFE) {
+                        NCDFE.printStackTrace();
+                        continue;
+                    } catch (Exception exception) {
+                        exception.printStackTrace();
+                        continue;
+                    }
+                }
+            }
+        }
+
+
+        writer = new PrintWriter(rootFold + "/" + num + ".txt", "UTF-8");
+        for (String key : nameToInstance.keySet()) {
+            writer.println(key);
+        }
+        writer.close();
+    }
+
+    /**
+     * Adds the current loadable classes to the current class list in 2tmp.txt file in the phase 2tmp.
+     * @throws IOException
+     */
+    public void capture_class() throws IOException {
+
+        String phase = readFile(MainAgent.tmpfile);
+
+        // read whitelist;
+        try (BufferedReader br = new BufferedReader(new FileReader(MainAgent.pkgFile))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                whiteList.add(line);
+            }
+        }
+        catch (Exception e) {
+            return;
+        }
+
+        // get all loadable classes
+        Class[] loadedClasses = MainAgent.getInstrumentation().getAllLoadedClasses();
+        Set<String> classes = new HashSet<>();
+        for (Class c : loadedClasses) {
+            String clz = c.getName();
+            if ((clz.contains("java.") && !clz.startsWith("java.lang.System"))
+                    || (clz.contains("javax.") && !clz.startsWith("javax.cache.Caching"))
+                    || clz.contains("javafx.")
+                    || clz.contains("jdk.")
+                    || clz.contains("scala.")
+                    || clz.contains("sun.")
+                    || clz.contains("edu.illinois.cs")
+                    || clz.contains("org.custommonkey.xmlunit")
+                    || clz.contains("org.junit")
+                    || clz.contains("statecapture.com.")
+                    || clz.contains("statecapture.org.")) {
+                continue;
+            }
+            classes.add(clz);
+        }
+
+        // write to the 2tmp.txt file
+        PrintWriter writer = new PrintWriter(rootFold + "/" + "2tmp.txt", "UTF-8");
+        for (String str : classes) {
+            writer.println(str);
+        }
+        writer.close();
+
+    }
+
+    protected boolean shouldCapture(Field f) {
+        return true;
+    }
+
+    String createSubxmlFold() {
+        int subxmlDirCnt = 0;
+        File f = new File(subxmlFold);
+        File[] files = f.listFiles();
+        if (files != null)
+            for (int i = 0; i < files.length; i++) {
+                if (files[i].isDirectory()) {
+                    subxmlDirCnt++;
+                }
+            }
+
+        String subxmlDir = subxmlFold + "/" + subxmlDirCnt +"xml";
```

Comment:
Should we let StateCapture come up with directory structure and names itself, or simply have the outside driver provide all this information to it?